ddtrace/commands/ddtrace_run.py

```diff
@@ -16,54 +27,89 @@
 log = logging.getLogger(__name__)
 
 USAGE = """
-Execute the given Python program after configuring it to emit Datadog traces.
-
-Append command line arguments to your program as usual.
+Execute the given Python command after configuring it to emit Datadog traces.
+The Datadog profiler can also be enabled.
 
-Usage: ddtrace-run <my_program>
-"""  # noqa: E501
 
-
-def _ddtrace_root():
-    from ddtrace import __file__
-    return os.path.dirname(__file__)
+Examples
+ddtrace-run python app.py
+ddtrace-run uwsgi app.py
+ddtrace-run gunicorn myproject.wsgi
+"""
 
 
 def _add_bootstrap_to_pythonpath(bootstrap_dir):
     """
     Add our bootstrap directory to the head of $PYTHONPATH to ensure
     it is loaded before program code
     """
-    python_path = os.environ.get('PYTHONPATH', '')
+    python_path = os.environ.get("PYTHONPATH", "")
 
     if python_path:
-        new_path = '%s%s%s' % (bootstrap_dir, os.path.pathsep, os.environ['PYTHONPATH'])
-        os.environ['PYTHONPATH'] = new_path
+        new_path = "%s%s%s" % (bootstrap_dir, os.path.pathsep, os.environ["PYTHONPATH"])
+        os.environ["PYTHONPATH"] = new_path
     else:
-        os.environ['PYTHONPATH'] = bootstrap_dir
+        os.environ["PYTHONPATH"] = bootstrap_dir
 
 
 def main():
-    if len(sys.argv) < 2 or sys.argv[1] == '-h':
-        print(USAGE)
-        return
-
-    log.debug('sys.argv: %s', sys.argv)
-
-    root_dir = _ddtrace_root()
-    log.debug('ddtrace root: %s', root_dir)
-
-    bootstrap_dir = os.path.join(root_dir, 'bootstrap')
-    log.debug('ddtrace bootstrap: %s', bootstrap_dir)
+    parser = argparse.ArgumentParser(
+        description=USAGE,
+        prog="ddtrace-run",
+        usage="ddtrace-run <your usual python command>",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument("command", nargs=argparse.REMAINDER, type=str, help="Command string to execute.")
+    parser.add_argument("-d", "--debug", help="enable debug mode (disabled by default)", action="store_true")
+    parser.add_argument("-i", "--info", help="print library info useful for debugging", action="store_true")
+    parser.add_argument("-p", "--profiling", help="enable profiling (disabled by default)", action="store_true")
+    parser.add_argument("-v", "--version", action="version", version="%(prog)s " + ddtrace.__version__)
+    args = parser.parse_args()
+
+    if args.profiling:
+        os.environ["DD_PROFILING_ENABLED"] = "true"
+
+    if args.debug:
+        os.environ["DD_TRACE_DEBUG"] = "true"
+
+    if args.info:
+        # Inline imports for performance.
+        import pprint
+        from ddtrace.internal.debug import collect
+
+        pprint.pprint(collect(ddtrace.tracer))
```

Comment:
what does this look like? is this the same as the startup logging?

Reply:
Yes, it's just the pretty-printed startup logs json.

```
~/d/dd-trace-py ❯❯❯ ddtrace-run --info
{'agent_error': 'Agent not reachable. Exception raised: [Errno 61] Connection '
                'refused',
 'agent_url': 'http://localhost:8126',
 'analytics_enabled': False,
 'architecture': '64bit',
 'in_virtual_env': True,
 'version': '0.43.0.dev9+g61c0e3f2',
 'vm': 'CPython'}
 'integrations': {'aiobotocore': {'config': None,
                                  'enabled': False,
                                  'instrumented': False,
                                  'module_available': True,
                                  'module_imported': False,
                                  'module_version': '1.1.0'},
...
```